ui/app/components/secret-edit.js

```diff
@@ -235,6 +240,51 @@ export default Component.extend(FocusOnInsertMixin, {
       set(this.modelForData, 'secretData', this.secretData.toJSON());
     },
 
+    handleWrapClick() {
+      this.set('isWrapping', true);
+      if (this.isV2) {
+        this.store
+          .adapterFor('secret-v2-version')
+          .queryRecord(this.modelForData.id, { wrapTTL: 1800 })
+          .then(resp => {
+            this.set('wrappedData', resp.wrap_info.token);
+            this.flashMessages.success('Secret Successfully Wrapped!');
+            this.set('showWrapButton', false);
+            this.set('isWrapping', false);
+          })
+          .catch(() => {
+            this.flashMessages.error('Could Not Wrap Secret');
+          });
+      } else {
+        this.store
+          .adapterFor('secret')
+          .queryRecord(null, null, { backend: this.model.backend, id: this.modelForData.id, wrapTTL: 1800 })
+          .then(resp => {
+            this.set('wrappedData', resp.wrap_info.token);
```

Comment:
Does this get reset or nulled out somewhere? Just thinking through how/why it doesn't stay through multiple popup openings.